client/js/lounge.js

```diff
@@ -742,13 +757,29 @@ $(function() {
 		}
 	}
 
-	sidebar.on("click", ".chan, button", function() {
+	sidebar.on("click", ".chan, button", function(e, data) {
 		var self = $(this);
 		var target = self.data("target");
 		if (!target) {
 			return;
 		}
 
+		if (!data || data.pushState !== false) {
+			var state = {
+				view: {
+					"#connect": VIEWS.CONNECT,
+					"#settings": VIEWS.SETTINGS,
+					"#sign-in": VIEWS.SIGN_IN
+				}[target] || VIEWS.CHANNEL
+			};
+			switch (state.view) {
+			case VIEWS.CHANNEL:
```

Comment:
That `VIEWS` object is a level of indirection I'd prefer to avoid.
Could you switch to use strings instead please: `connect`, `settings`, `sign-in`, `channel`?